k9mail/src/main/java/com/fsck/k9/activity/RecipientAdapter.java

```diff
@@ -0,0 +1,166 @@
+        holder.email.setText(address);
+
+        setContactPhotoOrPlaceholder(context, holder.photo, recipient);
+
+        if (!showCryptoStatus) {
+            holder.cryptoStatus.setVisibility(View.GONE);
+            return;
+        }
+        holder.cryptoStatus.setVisibility(View.VISIBLE);
+
+        Integer cryptoStatusRes = null, cryptoStatusColor = null;
+        RecipientCryptoStatus cryptoStatus = recipient.getCryptoStatus();
+        switch (cryptoStatus) {
+            case AVAILABLE_TRUSTED:
+                cryptoStatusRes = R.drawable.status_lock_closed;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_green);
+                break;
+            case AVAILABLE_UNTRUSTED:
+                cryptoStatusRes = R.drawable.status_lock_error;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_orange);
+                break;
+            case UNAVAILABLE:
+                cryptoStatusRes = R.drawable.status_lock_open;
+                cryptoStatusColor = context.getResources().getColor(R.color.openpgp_red);
+                break;
+        }
+
+        if (cryptoStatusRes != null) {
+            // we could do this easier with setImageTintList, but that's API level 21
+            Drawable drawable = context.getResources().getDrawable(cryptoStatusRes);
+            // noinspection ConstantConditions, we know the resource exists!
+            drawable.mutate();
+            drawable.setColorFilter(cryptoStatusColor, Mode.SRC_ATOP);
+            holder.cryptoStatus.setImageDrawable(drawable);
+            holder.cryptoStatus.setVisibility(View.VISIBLE);
+        } else {
+            holder.cryptoStatus.setVisibility(View.GONE);
+        }
+
+    }
+
+    public static void setContactPhotoOrPlaceholder(Context context, ImageView imageView, Recipient recipient) {
+        imageView.setImageDrawable(null);
+        if (recipient.photoThumbnailUri != null) {
+            Glide.with(context).load(recipient.photoThumbnailUri).into(imageView);
+        } else {
+            ContactPicture.getContactPictureLoader(context).loadContactPicture(recipient.address, imageView);
+        }
+    }
+
+    @Override
+    public Filter getFilter() {
+        return new Filter() {
+            @Override
+            protected FilterResults performFiltering(CharSequence constraint) {
+                if (recipients == null) {
+                    return null;
+                }
+                FilterResults result = new FilterResults();
+                result.values = recipients;
+                result.count = recipients.size();
+                return result;
+            }
+
+            @Override
+            protected void publishResults(CharSequence constraint, FilterResults results) {
+                notifyDataSetChanged();
+            }
+        };
+    }
+
+    static class RecipientTokenHolder {
+        TextView name, email;
```

Comment:
Code style:
* one variable/field declaration per line
* fields can be final
* fields should probably be public to indicate they are part of the "API"